### PR TITLE
THE BIG SCARY REPO MERGE

### PR DESCRIPTION
**Pull request by sage-wright**
# THE BIG SCARY REPO MERGE
### This PR is huge. It does many, many things.

Primarily, this PR combines, consolidates, squishes, and explodes 💥 all of the tasks in the PHBG and PHVG repositories into PHB. Unused tasks have been deleted (save for any species-specific modules we may add to TheiaProk in the future). Unused workflows have been deleted. Workflows that we will no longer provide active support for in this repository are deleted. 

#### Workflows have been organized within the workflows directory into sub-categories:
- `freyja` - contains the four Freyja workflows from PHVG
- `phylogenetics` - contains Core_Gene_SNP, kSNP3, MashTree_FASTA, all four workflows used in Augur, and Snippy_Tree
- `standalone_modules` - contains AMRFinderPlus, Gambit_Query, Kleborate, the two Krakens, Nullarbor, SerotypeFinder, the two TBProfilers, and Snippy_Variants
- `submission` - contains Mercury_Prep_N_Batch and Terra_2_NCBI
- `theiacov` - contains the five TheiaCoV workflows plus TheiaCoV_Validate
  - a subdirectory `updates` contains the two NCBI_Scrubs, Pangolin_Update, and VADR_Update
- `theiaeuk` - contains TheiaEuk_PE and Cauris_Cladetyper
- `theiaprok` - contains the ~~two~~ _four_ TheiaProk workflows
- `utilities` - contains the read_QC_trim subworkflows, merlin_magic (merges TheiaEuk-specific modules into the PHBG one), and a new subworkflow that performs all of the ivar primer trimming/consensus/variant calling tasks
  - a subdirectory `data_import` contains Basespace_Fetch, SRA_Fetch, and Terra_2_BQ
  - a subdirectory `file_handling` contains the Concatenate, Transfer, and Zip_Column_Content workflows

#### Tasks have been organized into eight sub-categories:
- `alignment` - contains bwa and mafft
- `assembly` - contains half of the exploded `task_ont_medaka.wdl` file (now artic_consensus), irma, shovill, and the entire exploded `task_consensus_call.wdl` file (now ivar_consensus, ivar_primer_trim, and ivar_variant_call)
- `gene_typing` - contains all PHBG gene_typing modules in addition to snippy_variants and sc2_gene_coverage
- `phylogenetic_inference` - contains all the PHBG phylogenetic_inference modules in addition to an exploded snp_dists.wdl (now separated into regular snp_dists and reorder_matrix), nextstrain_augur, and gubbins.
- `quality_control` - contains all of the PHBG and PHVG quality_control modules; duplicates were merged. Contains half of the exploded `task_ont_medaka.wdl` file (now artic_guppyplex), assembly_metrics, consensus_qc, etc. It also squishes several of the tasks to no longer require separate _pe and _se versions.
- `species_typing` - contains all of the PHBG species_typing modules; contains quasitools and pangolin as well. Several tasks were squished to no longer require separate _pe and _se versions, and those with standalone versions used in the `standalone_modules` workflow folder have been condensed if possible
- `taxon_id` - contains all PHBG taxon_id modules and the exploded infamous `taxon_id.wdl` file from PHVG. Includes Freyja, gambit, kraken2 (which no longer has separate _pe/_se versions, but does still contain a standalone version that does not have a default database), midas, and nextclade
- `utilities` - contains all previous PHBG utilities, and merges TheiaEuk-specific modules into previously existing ones; contains file manipulations for augur, contains export_taxon_table, ~~mercury_file_wrangling~~, nullarbor, rasusa, the phylogenetic summarize_data, theiacov_summary, and validate (used in theiacov_validate).
  - a subdirectory `submission` contains the mercury_file_wrangling, submission and broad_ncbi_tools task files

`task_versioning` is not in any subdirectory.

#### Other changes:
- All workflows in the repository (37!!!) have been added to the .dockstore.yml; all workflows have been given the suffix of `_PHB`
  - I'm not sure if Cauris_Cladetype_PHB is intended to be on Dockstore since it was not present in the PHB v0.0.1 validation workspace. I've added it anyways.
- All workflows have had their import paths adjusted accordingly and the new organization of files will not be an issue.
- The README.md has been initialized based off a copy of the PHVG README. 
- A CITATION.cff has been added; not sure if order here matters ~~but likely needs to be updated (currently missing Inês)~~ 
- I tried to add the `disk_size` variable for Azure compatibility to all tasks, but I may have missed some.
- All 37 workflows have been tentatively tested (I've made quite a few changes on this branch since my initial tests) and you can find appropriate data (nearly all of the validation data used for all three repos) for the workflows [in this Terra workspace](https://app.terra.bio/#workspaces/theiagen-validations/PHB-merge).
- a couple typos have been fixed, all workflows now show the PHB version (currently `PHB v0.1.0: merge`)
- Syntax has been (mostly) standardized; ~~(I haven't touched Augur and highly suggest no one does. I'm going to dig into Augur maybe in March)~~ I am currently digging into Augur.

There may be a few small remnants of old repositories that need to be removed, but they will require eagle eyes 🦅 

### TL;DR: nearly every single file has been changed 😁

### CODE REVIEW

**Comment by sage-wright** [comment]
@rpetit3 tagging you for GHA

**Comment by rpetit3** [comment]
Is there a rush to get this PR merged? Kind of considering if GHA should wait till after this PR, or wait till the GHA are set to merge this PR

**Comment by sage-wright** [comment]
Not 100% sure. We can probably wait until after this PR is merged for GHA

**Review by kevinlibuit — APPROVED** [review]

**Comment by kevinlibuit** [comment]
Rescinding approval for now. Will give a second review once @kapsakcj is able to update all default docker tags to use quay

**Comment by rpetit3** [comment]
> Is there a rush to get this PR merged? Kind of considering if GHA should wait till after this PR, or wait till the GHA are set to merge this PR

Following up on this, want GHA to go into this or wait till merge? 

Also, is `main` being used by others on Terra?

**Comment by kevinlibuit** [comment]
@rpetit3 I'm finishing up this review now hoping to merge to main tonight. Would love to see the GHA as a separate PR after things are set on `main`. RE use on Terra: most are either using a version release or one of the dev branches

**Review by kevinlibuit — APPROVED** [review]

**Comment by rpetit3** [comment]
I'll get the GHA knocked out this week
